configure.ac

```diff
@@ -339,36 +339,29 @@ EXTRA_WARNINGS=""
 
 WARNLIST="
 	all
+	extra
+	unused
 	shadow
 	missing-prototypes
 	missing-declarations
+	suggest-attribute=noreturn
+	suggest-attribute=format
 	strict-prototypes
-	declaration-after-statement
 	pointer-arith
 	write-strings
 	cast-align
 	bad-function-cast
 	missing-format-attribute
+	float-equal
 	format=2
-	format-security
-	format-nonliteral
-	no-long-long
-	unsigned-char
-	gnu89-inline
-	no-strict-aliasing
-	error
-	address
-	cpp
-	overflow
-	parentheses
-	sequence-point
-	switch
```

Comment:
Most of these are enabled by -Wextra. A few have no documentation on the GCC website and I couldn't find anything on them via stack overflow or google. If they're important, please feel free to re-enable.